cmd/main.go

```diff
@@ -49,11 +50,15 @@ func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var probeAddr string
+	var gardenerKubeconfigPath string
+
 	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	flag.StringVar(&gardenerKubeconfigPath, "gardener-kubeconfig-path", "/gardener/kubeconfig/kubeconfig", "Kubeconfig file for Gardener cluster")
```

Comment:
It is not clear from the deployment yaml, but it works as follows:
- the secret is mounted to the /gardener/kubeconfig directory
- the secret contains `kubeconfig` key, so that file with `kubeconfig` name is created in the mounted directory

What could be done:
- directory path may be changed to `/gardener/` or `/gardener/credentials`
- specific key from the secret may be mounted to make it more explicit

Reply:
`/gardener/credentials` would be a better name in my opinion, but after your explanation, I don't think it's obligatory to update that. I would leave it up to you.